docs/user_guide/configuring.rst

```diff
@@ -277,6 +277,93 @@ at the bottom. You can hide these buttons with the following configuration:
    }
 
 
+Add a dropdown to switch between docs versions
+==============================================
+
+You can add a button to your site's navbar or sidebars that allows users to
+switch between versions of the documentation. The versions shown in the
+dropdown are populated from a JSON file containing a list of mappings, where
+each mapping contains a key ``"version"`` mapped to a string value representing
+the version number. Each mapping may optionally contain a key ``"name"`` mapped
+to a name for that version (e.g., "latest", "stable", "dev", etc). Here is an
+example:
+
+.. code:: json
+
+    [
+        {
+            "name": "v2.1 (stable)",
+            "version": "2.1"
+        },
+        {
+            "version": "2.0"
+        },
+        {
+            "version": "1.1"
+        },
+    ]
+
+The location of this JSON file should be included in ``conf.py`` as the
+``switcher_json_url`` value within the ``html_context`` dictionary.
+Additionally, you must provide a *template URL* that will yield the correct
+address for the doc versions once the version string is inserted into the
+template in place of the ``{version}`` placeholder. Finally, specify in
+``switcher_version_match`` a value that the version entries in the JSON file
+will be matched against (typically this will be the sphinx variable ``version``
+or ``release`` depending on how granular your docs versioning is; see
+`the Sphinx documentation <https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information>`__
+for more info). An example configuration would be:
+
+.. code:: python
+
+    version = my_package_name.__version__.replace("dev0", "")  # may differ
+    html_context = {
```

Comment:
why is this `html_context` instead of `html_theme_options`? I think we might want `html_theme_options`

Reply:
moving the keys to `html_theme_options` did not work in my local testing today, so I've left them in `html_context` and removed the entries in `theme.conf`

Reply:
what do you mean by "did not work"? I feel like `html_theme_options` is the right mechanism for exposing theme-specific functionality, so maybe we are just not using them in the proper place in the templates or something?

Reply:
by "didn't work" I mean that I was doing it wrong  :)  I have it working locally now, will push soon